common/src/main/java/com/wynntils/utils/mc/type/PreciseLocation.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright © Wynntils 2024.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.utils.mc.type;
+
+import java.util.Comparator;
+import java.util.Objects;
+import net.minecraft.core.Position;
+import net.minecraft.world.phys.Vec3;
+
+public class PreciseLocation implements Position, Comparable<PreciseLocation> {
```

Comment:
Can you rename this `PrecisePosition` instead? I've tried hard to maintain the difference `Location` (meaning an integer value of a block) vs `Position` (meaning a real value of an entity) in the code base.